From ee4565b8676c478a3b70c6fe351832146715fa8b Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (Red Hat)" <rostedt@goodmis.org>
Date: Tue, 15 Nov 2016 16:34:21 -0500
Subject: [PATCH 108/134] trace-cmd: Free debug_str on error exit in
 tracecmd_find_tracing_dir()

On failing to mount the debugfs, free the debug_str before exiting the
function.

Signed-off-by: Steven Rostedt <rostedt@goodmis.org>
---
 trace-util.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/trace-util.c b/trace-util.c
index 14656d9aaaa7..6a0818d3263e 100644
--- a/trace-util.c
+++ b/trace-util.c
@@ -739,6 +739,7 @@ char *tracecmd_find_tracing_dir(void)
 			} else {
 				if (mount_debugfs() < 0) {
 					warning("debugfs not mounted, please mount");
+					free(debug_str);
 					return NULL;
 				}
 				strcpy(fspath, DEBUGFS_PATH);
-- 
2.5.5

